mimesis/builtins/pl.py

```diff
@@ -0,0 +1,106 @@
+"""Specific data provider for Poland (pl)."""
+
+from calendar import isleap
+from typing import Optional
+
+from mimesis.builtins.base import BaseSpecProvider
+from mimesis.enums import Gender
+from mimesis.providers import Datetime
+from mimesis.typing import DateTime
+
+__all__ = ['PolandSpecProvider']
+
+
+class PolandSpecProvider(BaseSpecProvider):
+    """Class that provides special data for Poland (pl)."""
+
+    class Meta:
+        """The name of the provider."""
+
+        name = 'poland_provider'
+
+    def nip(self) -> str:
+        """Generate random valid 10-digit NIP.
+
+        :return: valid 10-digit NIP
+        """
+        nip_digits = [int(d) for d in str(self.random.randint(101, 998))]
+        for _ in range(6):
+            nip_digits.append(self.random.randint(0, 9))
+        nip_coefficients = (6, 5, 7, 2, 3, 4, 5, 6, 7)
+        sum_v = sum(
+            map(lambda x: x[0] * x[1], zip(nip_coefficients, nip_digits)))
+        checksum_digit = sum_v % 11
+        if checksum_digit > 9:
+            checksum_digit = 0
+        nip_digits.append(checksum_digit)
+        return ''.join(str(d) for d in nip_digits)
+
+    def pesel(self,
+              birth_date: Optional[DateTime] = None,
+              gender: Optional[Gender] = None) -> str:
+        """Generate random 11-digit PESEL.
+
+        :param birth_date: initial birth date (optional)
+        :param gender: gender of person
+        :return: valid 11-digit PESEL
+        """
+        if not birth_date:
+            year = Datetime().year(1800, 2299) % 100
+            month = self.random.randint(1, 12)
+            if month in (4, 6, 9, 10):
+                day = self.random.randint(1, 30)
+            elif month == 2:
+                if isleap(year):
+                    day = self.random.randint(1, 29)
+                else:
+                    day = self.random.randint(1, 28)
+            else:
+                day = self.random.randint(1, 31)
+        else:
+            year = birth_date.date().year % 100
+            month = birth_date.date().month
+            day = birth_date.date().day
+        pesel_digits = list(map(int, '{:02d}'.format(year)))
+        if 1800 <= year <= 1899:
+            month += 80
+        elif 1900 <= year <= 1999:
```

Comment:
I agree with original review from @lk-geimfari "Why do we check it?". :) You misunderstood him then - he didn't ask for PESEL algorithm, but why do you check case for 1900-1999 if you `pass` it in next line. :) Why not just delete those two lines?

Reply:
@pywkm awesome! Thank you for your review.